Base/WebService.swift

```diff
@@ -266,13 +318,9 @@ fileprivate extension URLRequest {
             httpBody = data
         }
         
-        if let provider = resource.headerProvider {
-            for (header, value) in provider.headers() {
-                addValue(value, forHTTPHeaderField: header)
-            }
-        }
+        let headers = resource.headerProvider?.headers() ?? [] + additionalHeaders
```

Comment:
I find this line isn't clear, it's not common / easy to know the operator precedence of ?? and + so I think we should add some brackets

Reply:
Can add brackets for clarity if needed, but if the precedent was wrong we'd actually get a compilation error.